Sources/PackageModel/Toolset.swift

```diff
@@ -0,0 +1,214 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2014-2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+import class Basics.ObservabilityScope
+import struct TSCBasic.AbsolutePath
+import protocol TSCBasic.FileSystem
+import struct TSCBasic.RelativePath
+import struct TSCBasic.StringError
+import struct TSCUtility.Version
+
+/// A set of paths and flags for tools used for building Swift packages. This type unifies pre-existing assorted ways
+/// to specify these properties across SwiftPM codebase.
+public struct Toolset {
+    public enum KnownTool: String, Hashable, CaseIterable {
+        case swiftCompiler
+        case cCompiler
+        case cxxCompiler
+        case linker
+        case librarian
+        case debugger
```

Comment:
Should testing tools / XCTest also be part of this?

Reply:
What specific testing tools are we even allowing to be configurable?

Reply:
`ToolchainConfiguration` has an `xctestPath`

Reply:
Good catch, I've added `testRunner` to cover that.